newdle/client/src/components/summary/SummaryPage.js

```diff
@@ -36,14 +54,52 @@ export default function SummaryPage() {
     }
   };
 
-  const sendResultEmails = () => _sendResultEmails(newdle.code);
+  const handleMailModalClose = useCallback(() => {
+    setMailModalOpen(false);
+  }, [setMailModalOpen]);
+
+  const handleMailModalConfirm = useCallback(() => {
+    setMailModalOpen(false);
+    _sendResultEmails(newdle.code);
+  }, [setMailModalOpen, newdle, _sendResultEmails]);
 
   if (!newdle) {
     return <Loader active />;
   }
 
   const mailSent = sendMailResponse !== null;
 
+  const recipientList = (list, color, icon, limit, limitSetter) => {
+    var sliced_list = list.slice(0, limit);
```

Comment:
`var` is a no-no nowadays (always use `const` or `let` depending on whether you reassign or not)